Macro documentation in filter engine (#6392)
inverse-inc/packetfence: docs/installation/advanced_access_configuration.asciidoc

=== Filter Engine Macro ===

In each filter engines you have the possibility to use some macro in the text field:

    uc
    lc
    join
    substr
    macToEUI48
    random_from_range
    log
    replace
    BuildFromMatch

==== uc ====

This macro return the upper case of a string

Example:

  PacketFence-UserName = ${uc($radius_request.Calling-Station-Id)}

will assign to PacketFence-UserName the upper case value of Calling-Station-Id.

 Calling-Station-Id = "00:10:7f:38:89:9d" -> PacketFence-UserName = "00:10:7F:38:89:9D"

==== lc ====

This macro return the lower case of a string

Example:

  PacketFence-UserName = ${lc($radius_request.User-Name)}

will assign to PacketFence-UserName the lower case value of User-Name.

 User-Name = "ZAMMIT" -> PacketFence-UserName = "zammit"

==== join ====

This macro return the join of a string

Example:

  PacketFence-UserName = ${join(":",$radius_request.User-Name,"Super")}

will assign to PacketFence-UserName a join of the separator and the values.

 User-Name = "bobey" -> PacketFence-UserName = "bobey:Super"

==== substr ====

> This macro will return an substring

Reviewer: ```suggestion
This macro will return a substring
```

 PacketFence-UserName = ${substr($radius_request.User-Name,0, 5)}

will assign to PacketFence-UserName the substring

 User-Name = "ZammitLudovic" -> PacketFence-UserName = "Zammit"

==== macToEUI48 ====

> This macro will return the EUI48 format of a mac address

Reviewer: ```suggestion
This macro will return the EUI48 format of a MAC address
```

 PacketFence-UserName = ${macToEUI48($radius_request.Calling-Station-Id)}

> will assign to PacketFence-UserName the EUI48 mac address

Reviewer: ```suggestion
will assign to PacketFence-UserName the EUI48 MAC address
```

 Calling-Station-Id = "00:10:7f:38:89:9d" -> PacketFence-UserName = "00-10-7F-38-89-9D"

==== random_from_range ====

This macro will return a random number from a range

 Session-Timeout  = ${random_from_range("10620..12600")}

> will assign a number between 10620 and 12600 to Session-Timeout

Reviewer: ```suggestion
will assign a random number between 10620 and 12600 to Session-Timeout
```

 Session-Timeout = 11343

==== log ====

This macro will log a message in the packetfence.log file

 PacketFence-UserName = ${log($radius_request.User-Name." logged")}

> will log the value of the radius request attribute User-Name and logged

Reviewer: ```suggestion
will log the value of the RADIUS request attribute User-Name and logged
```

 User-Name = "zammit" -> "Zammit logged"

==== replace ====

This macro will replace a string by another one

 PacketFence-UserName = ${replace($radius_request.User-Name,"z","r")}

will replace the letter z by the letter r in the string

 User-Name = "zabbit" -> PacketFence-UserName = "rabbit"

==== BuildFromMatch ====

> This macro will apply a regex on the attribute value to be able able to build anither value

Reviewer: ```suggestion
This macro will apply a regular expression on the attribute value to be able able to build another value
```

  TLS-Stripped-UserName = ${BuildFromMatch($radius_request.TLS-Client-Cert-Common-Name,"^[^@]+","$0")}

will extract the value before the @ sign and assign this value

>  TLS-Client-Cert-Common-Name = "zammit@packetfence.org" -> TLS-Stripped-UserName = zammit

Reviewer: ```suggestion
 TLS-Client-Cert-Common-Name = "zammit@packetfence.org" -> TLS-Stripped-UserName = "zammit"
```
